src/models/ctx/update_profile.rs

```diff
@@ -30,6 +30,14 @@ pub fn update_profile<E: Env + 'static>(
                 Effects::none().unchanged()
             }
         }
+        Msg::Action(Action::Ctx(ActionCtx::DeleteAccount(password))) => match profile.auth_key() {
+            Some(auth_key) => Effects::one(delete_account::<E>(auth_key, password)).unchanged(),
```

Comment:
I think that Niki would add the UID to the action here as well as a guard in case that Delete ACcount was triggered with different account. 
I think it should be safe in this case as the API will return bad response if password doesn't mathc anyway.

Reply:
There is a guard there: https://github.com/Stremio/stremio-core/pull/753/files#diff-5e5b6a800019c0be2c32e04c71c27cda593eaef5d4b50e2a6772eb6e8cffac88R396
It's checking if the auth_key from the current profile is the same as the one that was sent in the request to prevent this